server/main.lua

```diff
@@ -225,14 +207,37 @@ RegisterNetEvent('qbx_vehicleshop:server:financePaymentFull', function(data)
 
     if not removeMoney(src, vehBalance) then return end
 
-    UpdateVehicleFinance({
+    require 'server.finance'.updateVehicleFinance({
         balance = 0,
         payment = 0,
         paymentsLeft = 0,
         timer = 0,
-    }, vehPlate)
+    }, vehId)
 end)
 
+---@param src number
+---@param data {coords: vector4, vehicleId: number, modelName: string, plate?: string}
+---@return number|nil
+local function spawnVehicle(src, data)
+    local coords, vehicleId = data.coords, data.vehicleId
+    local vehicle = vehicleId and exports.qbx_vehicles:GetPlayerVehicle(vehicleId) or data
+    if not vehicle then return end
+
+    local netId, veh = qbx.spawnVehicle({model = vehicle.modelName, spawnSource = coords, warp = GetPlayerPed(src)})
+
+    if not netId or netId == 0 then return end
+
+    if not veh or veh == 0 then return end
+
+    if vehicleId then Entity(veh).state:set('vehicleid', vehicleId, false) end
+
+    local plate = vehicle.plate or vehicle.props.plate
+    SetVehicleNumberPlateText(veh, plate)
```

Comment:
This should be passed as props into qbx.spawnVehicle to set the plate. This ensures the entity owner modifies the properties.